poulpe-service/src/main/java/org/jtalks/poulpe/logic/PermissionManager.java

```diff
@@ -159,6 +164,29 @@ public void changeRestrictions(Entity entity, PermissionChanges changes) {
     }
 
     /**
+     * Updates the principal and assign this one to the context.
+     *
+     * We need this functional because there are a rare cases when user details (like Id) of the principal
+     * are different from ones of current user in database. (It occurs when we cleared in database and
+     * then run the application with last saved session.)
+     *
+     * @param groups
+     */
+    private void updatePrincipal(List<Group> groups) {
+        User principal = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        for (Group group : groups) {
```

Comment:
What will if current user in not a member of not a member of any groups listed here?